doc/user/content/sql/drop-type.md

```diff
@@ -0,0 +1,46 @@
+---
+title: "DROP TYPE"
+description: "`DROP TYPE` removes a user-defined data type."
+menu:
+  main:
+    parent: 'sql'
+---
+
+{{< version-added v0.7.1 />}}
+
+`DROP TYPE` removes a user-defined [custom data type](../create-type). It cannot be used on default data types.
+
+## Syntax
+
+{{< diagram "drop-type.svg" >}}
+
+Field | Use
+------|-----
+_data_type_name_ | The name of the type to remove.
+`CASCADE` | Automatically drop objects that depend on the type (such as table columns, functions, operators, or other types)
```

Comment:
I don't believe this drops table columns, it would instead drop the entire table I believe. This also drops neither functions nor operators in practice because we don't allow you to define those objects currently. To get more context for what this does, you should chat with @JLDLaughlin.

Reply:
Sean is right! This would drop the entire table:
```
materialize=> create type int4_list as list (element_type = int4);
CREATE TYPE
materialize=> create table t (a int4_list);
CREATE TABLE
materialize=> drop type int4_list;
ERROR:  cannot drop materialize.public.int4_list: still depended upon by catalog item 'materialize.public.t'
materialize=> drop type int4_list cascade;
DROP TYPE
materialize=> select * from t;
ERROR:  unknown catalog item 't'
```